### PR TITLE
[buteo-sync-plugins] Use libcontacts SeasideContactBuilder helper for vCard

### PR DESCRIPTION
**Pull request by chriadam**
This commit reduces code duplication, and adds support for importing
PHOTO fields as contact avatars via Bluetooth / SyncML.

### CODE REVIEW

**Comment by chriadam** [comment]
Requires https://github.com/nemomobile/libcontacts/pull/98

File: storageplugins/hcontacts/ContactDetailHandler.cpp [hunk 1]
```diff
@@ -1,34 +1,189 @@
...
+#include <QContactSyncTarget>
+#include <QCryptographicHash>
+#include <QDir>
+#include <QImage>
+
+#include <qtcontacts-extensions.h>
+
+namespace {
+
+void processPhoto(const QVersitProperty &property, bool *alreadyProcessed, QList<QContactDetail> * updatedDetails)
```
Comment (matthewvogt) [inline storageplugins/hcontacts/ContactDetailHandler.cpp]: This is all duplicated from SeasidePropertyHandler?  If so, you could add a list of eligible detail types to SeasidePropertyHandler, and pass that in from the contact builder.

File: storageplugins/hcontacts/ContactsBackend.h [hunk 2]
```diff
@@ -223,7 +223,7 @@ class ContactsBackend
                                         (const QList<QContact> &aContactList);
 #endif
 
-    QList<QContact> convertVCardListToQContactList \
+    QList<QVersitDocument> convertVCardListToVersitDocumentList \
```
Comment (matthewvogt) [inline storageplugins/hcontacts/ContactsBackend.h]: Why is this line split?

**Comment by matthewvogt** [comment]
LGTM

File: storageplugins/hcontacts/ContactsBackend.h (hunk 2 above)
Comment (chriadam) [inline storageplugins/hcontacts/ContactsBackend.h]: It's Buteo code, originally from Nokia.  The coding style used throughout is... interesting.  I've been bad and mostly ignoring the coding style when I fix bugs or change implementation, which is why half the code has "Qt-style" and the other half has weird indentation, line splits, and almost-hungarian variable naming.

Comment (matthewvogt) [inline storageplugins/hcontacts/ContactsBackend.h]: There's no benefit to escaping the line-ending though, even if they want that coding style.  It probably just breaks compiler error messages...

**Pull request merged**